src/plans/conjugate_gradient_plan.jl

```diff
@@ -87,13 +89,14 @@ mutable struct ConjugateGradientDescentState{
         sC::StoppingCriterion,
         s::Stepsize,
         dC::DirectionUpdateRule,
+        res_cond::AbstractRestartCondition=RestartNever(),
```

Comment:
We could make this a bit nicer by writing `::RC` and adding `RC<:AbstractRestartcondition` in the `where` – then we can avoid the `typeof(...)` and just write `RC` there (also for the other other cases, just excluding `coef` and `βT`.